src/Symfony/Component/Asset/VersionStrategy/RemoteJsonManifestVersionStrategy.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\VersionStrategy;
+
+use Symfony\Contracts\HttpClient\Exception\ExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * Reads the versioned path of an asset from a remote JSON manifest file.
+ *
+ * For example, the manifest file might look like this:
+ *     {
+ *         "main.js": "main.abc123.js",
+ *         "css/styles.css": "css/styles.555abc.css"
+ *     }
+ *
+ * You could then ask for the version of "main.js" or "css/styles.css".
+ */
+class RemoteJsonManifestVersionStrategy implements VersionStrategyInterface
+{
+    private $manifestData;
+
+    /**
+     * @var ResponseInterface
+     */
+    private $httpResponse;
+
+    /**
+     * @param string $manifestUrl Absolute url to the manifest file
+     */
+    public function __construct(string $manifestUrl, HttpClientInterface $httpClient)
+    {
+        $this->httpResponse = $httpClient->request('GET', $manifestUrl, [
+            'headers' => ['accept' => 'application/json'],
+            'buffer' => true,
+        ]);
+    }
+
+    /**
+     * With a manifest, we don't really know or care about what
+     * the version is. Instead, this returns the path to the
+     * versioned file.
+     */
+    public function getVersion(string $path)
+    {
+        return $this->applyVersion($path);
+    }
+
+    public function applyVersion(string $path)
+    {
+        return $this->getManifestPath($path) ?: $path;
+    }
+
+    private function getManifestPath(string $path): ?string
+    {
+        if (null === $this->manifestData) {
+            try {
+                $this->manifestData = $this->httpResponse->toArray();
+            } catch (ExceptionInterface $e) {
+                throw new \RuntimeException(sprintf('Error loading asset manifest from url "%s"', $this->httpResponse->getInfo()['url']), 0, $e);
```

Comment:
why not let the original exception bubble down?

Reply:
That gives some added context, isn't it ?

Reply:
When the `toArray` is called on an invalid response, [`JsonException` message](https://github.com/symfony/symfony/blob/v5.0.4/src/Symfony/Component/HttpClient/Response/ResponseTrait.php#L156) is only `Syntax error`. That doesn't give any context on the URL that failed.